src/test/java/space/obminyashka/items_exchange/service/UserServiceTest.java

```diff
@@ -60,14 +59,28 @@ void setUp() {
     @Test
     void testLoginUserWithOAuth2_WhenUserBeenCreated() {
         var oauth2User = createDefaultOidcUser();
-        when(userRepository.findByEmailOrUsername(NEW_USER_EMAIL,NEW_USER_EMAIL)).thenReturn(creatOptionalUser());
+        when(userRepository.findByEmailOrUsername(NEW_USER_EMAIL, NEW_USER_EMAIL)).thenReturn(creatOptionalUser());
         userService.loginUserWithOAuth2(oauth2User);
 
         verify(userRepository).setOAuth2LoginToUserByEmail(oauth2UserArgumentCaptor.capture());
+
         assertEquals(NEW_USER_EMAIL, oauth2UserArgumentCaptor.getValue());
     }
 
-    private Optional<User> creatOptionalUser(){
+    @Test
+    void testLoginUserWithOAuth2_WhenUserBeenCreatedAndHasValidatedEmailWithOauth2Login() {
+        var oauth2User = createDefaultOidcUser();
+        var optionalUser = creatOptionalUser();
+        optionalUser.get().isValidatedEmail(true);
+        optionalUser.get().setOauth2Login(true);
+
+        when(userRepository.findByEmailOrUsername(NEW_USER_EMAIL, NEW_USER_EMAIL)).thenReturn(optionalUser);
+        userService.loginUserWithOAuth2(oauth2User);
+
+        verify(userRepository, never()).setOAuth2LoginToUserByEmail(oauth2UserArgumentCaptor.capture());
+    }
+
+    private Optional<User> creatOptionalUser() {
```

Comment:
Why don't you return a User instead of Optional and use Optional directly in the mocking places?